commons/src/test/java/org/infinispan/commons/util/MutableIntSetTest.java

```diff
@@ -0,0 +1,500 @@
+      intSet2.add(4);
+      intSet2.add(7);
+
+      assertFalse(intSet.containsAll(intSet2));
+      assertTrue(intSet2.containsAll(intSet));
+   }
+
+   @Test
+   public void testContainsAll1() throws Exception {
+      Set<Integer> intSet2 = new HashSet<>();
+
+      intSet.add(1);
+      intSet.add(4);
+
+      intSet2.add(1);
+      intSet2.add(4);
+      intSet2.add(7);
+
+      assertFalse(intSet.containsAll(intSet2));
+      assertTrue(intSet2.containsAll(intSet));
+   }
+
+   @Test
+   public void testAddAll() throws Exception {
+      intSet.add(1);
+      intSet.add(4);
+
+      IntSet intSet2 = new SmallIntSet();
+      intSet2.addAll(intSet);
+      assertEquals(2, intSet2.size());
+   }
+
+   @Test
+   public void testAddAll1() throws Exception {
+      IntSet rangeSet = new RangeSet(4);
+
+      intSet.addAll(rangeSet);
+      assertEquals(4, intSet.size());
+   }
+
+   @Test
+   public void testAddAll2() throws Exception {
+      Set<Integer> hashSet = new HashSet<>();
+      hashSet.add(1);
+      hashSet.add(4);
+
+      intSet.addAll(hashSet);
+      assertEquals(2, intSet.size());
+   }
+
+   @Test
+   public void testRemoveAll() throws Exception {
+      intSet.add(1);
+      intSet.add(4);
+      intSet.add(7);
+
+      assertEquals(3, intSet.size());
+
+      IntSet intSet2 = new SmallIntSet();
+      intSet2.add(4);
+      intSet2.add(5);
+      intSet2.add(7);
+
+      assertTrue(intSet.removeAll(intSet2));
+
+      assertEquals(1, intSet.size());
+      assertTrue(intSet.contains(1));
+   }
+
+   @Test
+   public void testRemoveAll1() throws Exception {
+      intSet.add(1);
+      intSet.add(4);
+      intSet.add(7);
+
+      assertEquals(3, intSet.size());
+
+      // (1 - 4)
+      IntSet rs = new RangeSet(5);
+
+      assertTrue(intSet.removeAll(rs));
+
+      assertEquals(1, intSet.size());
+      assertTrue(intSet.contains(7));
+   }
+
+   @Test
+   public void testRemoveAll2() throws Exception {
+      intSet.add(1);
+      intSet.add(4);
+      intSet.add(7);
+
+      assertEquals(3, intSet.size());
+
+      Set<Integer> hashSet = new HashSet<>();
+      hashSet.add(4);
+      hashSet.add(5);
+      hashSet.add(7);
+
+      assertTrue(intSet.removeAll(hashSet));
+
+      assertEquals(1, intSet.size());
+      assertTrue(intSet.contains(1));
+   }
+
+   @Test
+   public void testRetainAll() throws Exception {
+      intSet.add(1);
+      intSet.add(4);
+      intSet.add(7);
+
+      assertEquals(3, intSet.size());
+
+      IntSet intSet2 = new SmallIntSet();
+      intSet2.add(4);
+      intSet2.add(5);
+      intSet2.add(7);
+
+      intSet.retainAll(intSet2);
+
+      assertEquals(2, intSet.size());
+      assertTrue(intSet.contains(4));
+      assertTrue(intSet.contains(7));
+   }
+
+   @Test
+   public void testRetainAll1() throws Exception {
+      intSet.add(1);
+      intSet.add(4);
+      intSet.add(7);
+
+      assertEquals(3, intSet.size());
+
+      // (1 - 4)
+      IntSet rs = new RangeSet(5);
```

Comment:
Comment is a bit misleading, should be 0-4 maybe. IMO should add a method `hashSetOf(int...)` so you can simplify all the tests that require other sets, e.g.

```
assertTrue(intSet.retainAll(hashSetOf(1, 4));
assertEquals(hashSetOf(1, 4), intSet);
```

If we ever decide to stop implementing `Set<Integer>` in our custom sets, we'll just need a new `assertIntSetEquals` method :)

Reply:
Hrmm sure.

Reply:
Btw I can use `CollectionFactory.makeSet(T...)` method

Reply:
Sure, `CollectionFactory.makeSet()` sounds good. Sorry it took so long to respond, but initially I thought `CollectionFactory.makeSet()` would also delegate to our internal implementations, and I like the idea of depending only on JDK types in these tests.